pkg/apis/controller/v1alpha1/types_test.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2019 The GitLab-Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"testing"
+
+	xpcorev1alpha1 "github.com/crossplaneio/crossplane/pkg/apis/core/v1alpha1"
+	"github.com/go-test/deep"
+	"github.com/onsi/gomega"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplaneio/gitlab-controller/pkg/test"
+)
+
+const (
+	testNamespace = "default"
+	testName      = "test-gitlab"
+)
+
+var (
+	c client.Client
+
+	testKey = types.NamespacedName{
+		Name:      testName,
+		Namespace: testNamespace,
+	}
+
+	testMeta = metav1.ObjectMeta{
+		Name:      testName,
+		Namespace: testNamespace,
+	}
+)
+
+func TestMain(m *testing.M) {
+	t := test.NewEnv(testNamespace, SchemeBuilder.SchemeBuilder, test.CRDs())
+	c = t.StartClient()
+	t.StopAndExit(m.Run())
+}
+
+func TestStorageGitLab(t *testing.T) {
+	created := &GitLab{ObjectMeta: testMeta}
+	g := gomega.NewGomegaWithT(t)
+
+	// Test Create
+	fetched := &GitLab{}
+	g.Expect(c.Create(context.TODO(), created)).NotTo(gomega.HaveOccurred())
+
+	g.Expect(c.Get(context.TODO(), testKey, fetched)).NotTo(gomega.HaveOccurred())
+	g.Expect(fetched).To(gomega.Equal(created))
+
+	// Test Updating the Labels
+	updated := fetched.DeepCopy()
+	updated.Labels = map[string]string{"hello": "world"}
+	g.Expect(c.Update(context.TODO(), updated)).NotTo(gomega.HaveOccurred())
+
+	g.Expect(c.Get(context.TODO(), testKey, fetched)).NotTo(gomega.HaveOccurred())
+	g.Expect(fetched).To(gomega.Equal(updated))
+
+	// Test Delete
+	g.Expect(c.Delete(context.TODO(), fetched)).NotTo(gomega.HaveOccurred())
+	g.Expect(c.Get(context.TODO(), testKey, fetched)).To(gomega.HaveOccurred())
+}
+
+const (
+	readyType    = xpcorev1alpha1.Ready
+	creatingType = xpcorev1alpha1.Creating
+	deletingType = xpcorev1alpha1.Deleting
+	failedType   = xpcorev1alpha1.Failed
```

Comment:
I don't think we reference these constants anywhere beside the vars below. Why do we create local variables referencing the original constants declared in `xpcorev1alpha1`?

Reply:
Merely a convenience of not having to type too many characters :)